### ISSUE TITLE
1862 Refinancing Sequence Issue

### ISSUE THREAD

**Issue by ickyfoot** [tobymao/18xx#10037]
**How to Reproduce the Problem?**

I was prompted twice to Redeem the same share as part of a refi in 1862. I held the Director's Certificate, was prompted to sell/redeem that for half value, then afterwards I was prompted to sell/redeem another option for half value. I opted to redeem both times but was only charged once (which I understand to be correct per the rules). A clone to reproduce this is here:

https://18xx.games/hotseat/hs_jlinmpdc_145383

Screenshots are also attached.

**Which Browser / Platform?**
- Browser Version
  - Chrome: 120.0.6099.144 (Official Build) (64-bit) 
- Platform: Mobile, Android

**Screenshots of refi process:**
![Screenshot_20231221-175412](https://github.com/tobymao/18xx/assets/3941647/4b351fec-611a-4785-b104-a7f6f74c3710)
![Screenshot_20231221-175408](https://github.com/tobymao/18xx/assets/3941647/76bec619-ab04-42e7-9ada-eb55a0ae0d90)
![Screenshot_20231221-175403](https://github.com/tobymao/18xx/assets/3941647/a0774ce6-d837-4e01-b9a3-5c4604c27e7e)

**Screenshots of start end end personal cash**
![Screenshot_20231221-175848](https://github.com/tobymao/18xx/assets/3941647/92d8f800-3b3b-4303-bbcc-e8f815129de4)
![Screenshot_20231221-175830](https://github.com/tobymao/18xx/assets/3941647/dcd6f23f-edb5-4e0a-99c1-a9c34d1c4b09)

**Comment by ollybh** [comment]
@ickyfoot The link the the hotseat game you provided will only work on your computer, and only on the browser that  had the hotseat game.

Was this from a multiplayer game? Can you provide a link to the game? Copy the link from your browser which will look something like `https://18xx.games/game/31337`.

If not, can you send a link to your raw game data by using the `Tools` tab and `Copy Game Data`. Paste that data [into a github gist](https://gist.github.com/) and share the resulting link which will look something like `https://gist.github.com/[your_name]/[some_letters]`.

**Comment by ickyfoot** [comment]
It's from this game, which has since finished:
https://18xx.games/game/145383

Here's a link to the point in the game right before this happened:
https://18xx.games/game/145383?action=287

I ended up undoing and selling shares from the treasury...hopefully you can
clone and undo back to that same point in the game and refinance NGC to see
the issue in action.

**Comment by roseundy** [comment]
I took a look at this and I don't see any problems.

I didn't see the same share being redeemed twice. Instead, I saw the director's certificate owned by the president being redeemed and a share owned by the corporation being redeemed.

At the start of the refinance, this is what the shares of NGC look like:

ickyfoot - 5 shares (1 [3 share] director's certificate + 2 normal shares)
NGC - 5 shares in treasury

As part of the refinance operation, all shareholders have to place half of their NGC shares to the market.

ickyfoot has 5 shares, so they place 2 shares into the market. The director's cert can't be broken up, so ickyfoot receives a 2.5 share director's option and has to either redeem (buy) 1/2 a share (bringing their ownership to 3 shares - the full director's cert) for £34 or sell their 2.5 share option for £170. ickyfoot chooses to redeem for £34 out of their cash.

Now it's NGC's turn. It places 2 shares into the market and receives a 0.5 share option. It has to choose between redeeming the 0.5  share option or selling the 0.5 share option. NGC chooses to redeem for £34 out of it's treasury. Since ickyfoot is the director of NGC, ickyfoot is making these choices on behalf of NGC.

All this being done, NGC receives £821 for refinancing.

This is what the shares now look like:
ickyfoot - 3 shares (1 director's certificate)
NGC - 3 shares in treasury
Market - 4 shares

Like I said, this looks fine and I don't see a bug here.
